### ISSUE TITLE
Is offline likely to happen?

### ISSUE THREAD

**Issue by dabayliss** [pweingardt/mylyn-gitlab#19]
We have gitlab enterprise that requires me to be on a vpn to access the server. I like to be able to work WITHOUT the vpn active - which requires offline mode.

So - is offline likely to happen? Or not so much?

**Comment by samrocketman** [comment]
With git in general you can do that (I'm sure you're aware).  But I would see challenges around issue tracking in an offline mode.  Especially if you try to create new issues.  Though, I guess offline created issues can be queued somehow and given proper issue numbers when they're actually created.

**Comment by dabayliss** [comment]
The github mylyn connector handles this fine - it is really reading existing issues that is most critical. I use the context sensitivity on an activated issue - so I kinda need the issue to be there ...

**Comment by samrocketman** [comment]
Ah, I see what you mean.  So you're saying you want it to cache existing issues so you can read them offline.

**Comment by dabayliss** [comment]
yes - the github connect has a local store that it 'synchronizes' with the online one when it can. So we can all trundle along reading our issues (and actually updating them - we maybe get one conflict a year).

**Comment by Scibearspace** [comment]
I'm interested in this plugin and this problem in particular.
My solution would be to ask if it is possible to connect to 2 repositories simultaneously.
For instance in my work flow I have a local install of gitlab community, and push my code up to a 'private' space on gitlab.com
Is there a possibility of 'synchronising' between 2 servers simultaneously ?
To avoid the 'conflicting issues' problem.
can tasks / issues / comments etc be created with an 'alpha-numeric' value which can then be modified in the local once a value is given from the master repo (in my instance the master would be gitlab.com, or my local CE install).
I've had a look through the code but I'm not finding the class that pushes / synchronises to the remote repo. Or I would have a go at implementing myself.
help would be appreciated as this is something I really want to have (syncrhonisation between instances).

Thanks

David

**Comment by samrocketman** [comment]
@Scibearspace what you describe (tracking issues from two sources) is completely different than this request.  I recommend you create a new issue else risk your request never being addressed or tracked.

**Comment by n0mer** [comment]
@samrocketman any update on this?

**Comment by samrocketman** [comment]
Not that I’m aware.  Looking back at the comment history from 2 years ago I was in the peanut gallery commenting and not writing code for this feature.
